org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/codegen/BranchLabel.java

```diff
@@ -85,6 +85,29 @@ public void becomeDelegateFor(BranchLabel otherLabel) {
 	// other label is delegating to receiver from now on
 	otherLabel.delegate = this;
 
+	if (this.targetStackDepth == -1) {
```

Comment:
@srikanth-sankaran quick check - Should this be "< 0 " ?
[Essentially checking any scenario where this can be false]

Reply:
Not really, it is checking for the initial value from line 28 above. If you check all other assignments to this field, they are all guaranteed to be 0 or +ve